src/lambdaconnect_model/graph_generator.cljc

```diff
@@ -0,0 +1,145 @@
+                                                (map t/datomic-name)
+                                                (set))))
+
+        ;; Create objects, sort them by uuid and remove all relationships
+
+        objects (->> (range vertices) 
+                     (u/pmap (fn [_] (let [entity-name (rand-nth entity-names)
+                                           gen (gens entity-name)
+                                           generated (first (gen/sample gen 1))
+                                           interesting-keys (set (keys generated))] 
+                                       [entity-name 
+                                        (ref (select-keys generated 
+                                                          (difference interesting-keys 
+                                                                      (datomic-rels-for-entity entity-name))))
+                                        
+                                        ])))
+                     (group-by first)
+                     ((fn [x] 
+                        (update-vals x 
+                                     (fn [samples]
+                                       (into {} (map 
+                                                 #(vector (:app/uuid @%) %) 
+                                                 (map second samples))))))))
+        existing-relationships (ref #{})
+        
+        
+        add-relationship (fn [retry-count] 
+                           (let [entity-name (rand-nth entity-names)
+                                 entity (get entities-by-name entity-name)
+                                 relationship (->> entity 
+                                                  :datomic-relationships
+                                                  keys
+                                                  rand-nth
+                                                  (get (:datomic-relationships entity)))
+                                 result 
+                                 (when relationship 
+                                   (dosync
+                                    (let [rel-name (t/datomic-name relationship)
+                                          sources (-> objects
+                                                       (get entity-name))
+                                          source (-> sources vals rand-nth)
+                                          targets (-> objects
+                                                       (get (:inverse-entity relationship)))
+                                          target (-> targets vals rand-nth)]
+                                      (when (and source target)
+                                        (let [rel-flag [rel-name 
+                                                        (:app/uuid @source) 
+                                                        (:app/uuid @target)]]
+                                              (when-not (@existing-relationships rel-flag)
```

Comment:
Jeśli relacja `A->B` jest 1-1, wcześniej wylosowałem `rel-flag [A->B A1 B1]`, to czy po wylosowaniu `rel-flag [A->B A2 B1]` to nie zostanie przypadkiem przyjęte (i w `B1` nadpiszesz `A1` przez `A2` w relacji odwrotnej)?